src/Version.h

```diff
@@ -0,0 +1,133 @@
+/*
+* Copyright (C) 2021 German Aerospace Center (DLR/SC)
+*
+* Created: 2021-11-02 Martin Siggel <Martin.Siggel@dlr.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#ifndef VERSION_H
+#define VERSION_H
+
+#include <string>
+#include <regex>
+#include <exception>
+
+/**
+ * @brief A class to parse and compare version strings
+ */
+class Version
+{
+public:
+    explicit Version(const std::string& str)
+        : m_major(0),
+          m_minor(0),
+          m_patch(0)
+    {
+        parse(str);
+    }
+
+    int vMajor() const
+    {
+        return m_major;
+    }
+
+    int vMinor() const
+    {
+        return m_minor;
+    }
+
+    int vPatch() const
+    {
+        return m_patch;
+    }
+
+    std::string vLabel() const
+    {
+        return m_label;
+    }
+
+    std::string vBuild() const
+    {
+        return m_build;
+    }
+
+    bool operator< (const Version& other) const
+    {
+        if (vMajor() < other.vMajor()) {
+            return true;
+        }
+        else if (vMajor() > other.vMajor()) {
+            return false;
+        }
+
+        // major Versions are same
+        if (vMinor() < other.vMinor()) {
+            return true;
+        }
+        else if (vMinor() > other.vMinor()) {
+            return false;
+        }
+
+        // major and minor are same
+        if (vPatch() < other.vPatch()) {
+            return true;
+        }
+        else if (vPatch() > other.vPatch()) {
+            return false;
+        }
+
+        // we don't compare build or pre-release right now
+        // is there any sane way to do it???
```

Comment:
Yes, semver defines how to do this: https://semver.org/#spec-item-11:

> 4. Precedence for two pre-release versions with the same major, minor, and patch version MUST be determined by comparing each dot separated identifier from left to right until a difference is found as follows:
 >
 >    1. Identifiers consisting of only digits are compared numerically.
 >
 >    2. Identifiers with letters or hyphens are compared lexically in ASCII sort order.
 >
 >    3. Numeric identifiers always have lower precedence than non-numeric identifiers.
 >
 >    4. A larger set of pre-release fields has a higher precedence than a smaller set, if all of the preceding identifiers are equal.
 >
 > Example: 1.0.0-alpha < 1.0.0-alpha.1 < 1.0.0-alpha.beta < 1.0.0-beta < 1.0.0-beta.2 < 1.0.0-beta.11 < 1.0.0-rc.1 < 1.0.0.

Two versions that differ only in the build metadata have the same precedence.

Again, this is only a comment. I don't know if we need to implement this right now for the cpacs version check.

Reply:
Thanks for the hint. Although we don't need this right now, it could be good to implement this.

Reply:
Done